src/redux/slices/userSlice.js

```diff
@@ -0,0 +1,124 @@
+import { addBookmark, removeBookmark, addLike, removeLike } from "./postFunctions";
+
+import {
+    getUser, setUserInLocalStorage, addToFollowing,
+    addToFollowers, removeFromFollowers, removeFromFollowing, postProfileData
+} from "./userFunctions";
+
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import { getAuth, createUserWithEmailAndPassword, signInWithEmailAndPassword, signOut } from "@firebase/auth";
+import { getDocs, addDoc, arrayRemove } from "@firebase/firestore";
+import { usersRef } from "../../firebase";
+
+//logged in user
+const initialState = localStorage.getItem("user") === null ? {} : JSON.parse(localStorage.getItem("user"));
+
+const auth = getAuth();
+
+
+const addUserToDB = async (fullName, username, uid) => {
+    try {
+        const res = await addDoc(usersRef, {
+            ...initialState, fullName: fullName, username: username, uid: uid
+        });
+        setUserInLocalStorage(uid);
+    }
+    catch (e) {
+        console.log(e);
+    }
+}
+
+export const userSignUp = createAsyncThunk(
+    "user/signup",
+    async (arg) => {
+        try {
+            const res = await createUserWithEmailAndPassword(auth, arg.email, arg.password);
+            addUserToDB(arg.fullName, arg.username, res.user.uid)
+            return res.user;
+        } catch (e) { console.log(e) }
+    }
```

Comment:
Got a good idea of doing authentication through other services !! Another thing to learn about in Redux . Thanks Shivani☑